File: array/2022_convert_1d_array_into_2d_array/2022_convert_1d_array_into_2d_array.py
class Solution:
    def construct2DArray(self, original: List[int], m: int, n: int) -> List[List[int]]:
        result = []
        if len(original) != m*n:
            return result
        for i in range(m):
            result.append(original[i*n:(i+1)*n])
        return result

File: array/2022_convert_1d_array_into_2d_array/2022_convert_1d_array_into_2d_array.go
package main

func construct2DArray(original []int, m int, n int) [][]int {
	k := len(original)
	if k != m*n {
		return nil
	}
	ans := make([][]int, 0, m)
	for i := 0; i < k; i += n {
		ans = append(ans, original[i:i+n])
	}
	return ans

}
